runtime/oti/VMHelpers.hpp

```diff
@@ -2113,19 +2123,18 @@ class VM_VMHelpers
  *   another GC thread winning to scan(bit0/bit1,bit0 and bit1 is irrelevant and independent), again don't do anything, and let the winning thread do the work, instead
  */
 	static VMINLINE bool
-	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC)
-	{
-		uintptr_t complementGCConcurrentState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
-		uintptr_t returnedState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
-		do {
-			/* preserve the concurrent GC state for the other type of GC */
-			complementGCConcurrentState = continuation->state & getConcurrentGCMask(!isGlobalGC);
-			returnedState = VM_AtomicSupport::lockCompareExchange(&continuation->state, complementGCConcurrentState, complementGCConcurrentState | getConcurrentGCMask(isGlobalGC));
-		/* if the other GC happened to change its concurrentGC state since us taking a snapshot of their state, we'll have to retry */
-		} while (complementGCConcurrentState != (returnedState & complementGCConcurrentState));
-
-		/* if returned state does not contain carrier ID, return that we won */
-		return (complementGCConcurrentState == returnedState);
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC, bool beingMounted)
+	{
+		bool won = false;
+		volatile uintptr_t *localAddr = &continuation->state;
+		uintptr_t continuationState = *localAddr;
+		while (!won && ((!beingMounted && !isContinuationMountedOrConcurrentlyScanned(continuationState, isGlobalGC)) ||
```

Comment:
very dense code, getting harder to read...

how about keeping only !won condition for while loop, then inside the body of the while we have this:


bool shouldTry = !isConcurrentlyScannedFromContinuationState(continuationState, isGlobalGC):
if (!beingMounted) {
   shouldTry &= !isContinuationMounted(continuation);
}

if (shouldTry) {
   atomic;
}

a couple of comments embedded within the code are very welcome (we already had some, but lost)